src/web_interface/filter.py

```diff
@@ -421,3 +424,43 @@ def get_searchable_crypto_block(crypto_material: str) -> str:
     """crypto material plugin results contain spaces and line breaks -> get a contiguous block without those"""
     blocks = crypto_material.replace(' ', '').split('\n')
     return sorted(blocks, key=len, reverse=True)[0]
+
+
+def version_is_compatible(version: Union[str, semver.Version], other: Union[str, semver.Version]) -> bool:
+    """A warpper around ``semver.Version.is_compatible`` that allows non semver versions.
+    If only one of :paramref:`version` and :paramref:`other` is semver they are always considered incompatible.
+    So for example '1.1.0' would not be compatible '1.2' even if one might expand '1.2' to '1.2.0'.
```

Comment:
```suggestion
    """A wrapper around ``semver.Version.is_compatible`` that allows non semver versions.
    If only one of :paramref:`version` and :paramref:`other` is semver they are always considered incompatible.
    So for example '1.1.0' would not be compatible to '1.2' even if one might expand '1.2' to '1.2.0'.
```

Should we try expanding the versions in this function? We want to have to redo analyses as little as possible, so this should probably help?